### PR TITLE
Fix task property annotation problems

### PR DESCRIPTION
**Pull request by lptr**
This PR fixes some validation problems with tasks in the plugin that will cause any build using the plugin to fail with Gradle 7.0. It also enables strict validation during the build, so any new validation problems introduced will make the build fail.

The particular problems fixed by this PR are:

```text
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':asciidoctoreditorconfig:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.5.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Type 'org.asciidoctor.gradle.editorconfig.AsciidoctorEditorConfigGenerator': property 'additionalFileProviders' is missing a normalization annotation, defaulting to PathSensitivity.ABSOLUTE.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':asciidoctor-gradle-slides-export:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.5.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Type 'org.asciidoctor.gradle.slides.export.base.AbstractExportBaseTask': property 'outputDir' is annotated with @PathSensitive that is not allowed for @OutputDirectory properties.
   > Warning: Type 'org.asciidoctor.gradle.slides.export.decktape.DeckTapeTask': property 'outputDir' is annotated with @PathSensitive that is not allowed for @OutputDirectory properties.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':asciidoctor-gradle-jvm-pdf:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.5.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Type 'org.asciidoctor.gradle.jvm.pdf.AsciidoctorPdfTask': property 'fontsDir' is not annotated with an input or output annotation.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':asciidoctor-gradle-jvm-leanpub:validateTaskProperties'.
> Task property validation failed. See https://docs.gradle.org/5.5.1/userguide/more_about_tasks.html#sec:task_input_output_annotations for more information on how to annotate task properties.
   > Warning: Type 'org.asciidoctor.gradle.jvm.leanpub.DropboxCopyTask': property 'sourceDir' is missing a normalization annotation, defaulting to PathSensitivity.ABSOLUTE.
```

Also note that there are some validation warnings in the JRuby plugin that this plugin depends on. These warnings are fixed in https://github.com/jruby-gradle/jruby-gradle-plugin/pull/419. Once the JRuby plugin PR is merged, the dependency in the Asciidoctor plugin needs to be bumped in order to make this plugin fully compatible with Gradle 7.0.

### CODE REVIEW

**Comment by lptr** [comment]
Bumped to JRuby 2.0.1 after https://github.com/jruby-gradle/jruby-gradle-plugin/pull/419 has been mereged.

File: base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy [hunk 1]
```diff
@@ -638,6 +635,11 @@ abstract class AbstractAsciidoctorBaseTask extends DefaultTask {
         this.outDir = createDirectoryProperty(project)
     }
 
+    @Nested
```
Comment (ysb33r) [inline base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy:638]: @lptr Why is this change necessary for Gradle 7?

**Comment by ysb33r** [comment]
@lptr Can you base this against the development-3.x branch please?

File: base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy (hunk 1 above)
Comment (lptr) [reply base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy:638]: It is not allowed to annotate fields anymore. This is to replace the annotation on the `configuredOutputOptions` field on line 89.

Comment (ysb33r) [reply base/src/main/groovy/org/asciidoctor/gradle/base/AbstractAsciidoctorBaseTask.groovy:638]: Ah. Noted.

**Comment by lptr** [comment]
Thanks @ysb33r .
